### PR TITLE
Use new "true circle" tool from glue-jupyter

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is a companion PR to:

* https://github.com/glue-viz/glue-jupyter/pull/376

- [x] Bump glue-jupyter minversion when it is released with the required feature.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-3545)

### CODE REVIEW

**Comment by dhomeier** [comment]
Tried this with the latest update to glue-viz/glue-jupyter#376, working the same way as in my previous tests.
I.e. works as it should when moving the subregion, but on resizing it still snaps to an `EllipticalROI` as reported in https://github.com/glue-viz/glue-jupyter/pull/376#issuecomment-1642500383 .

**Comment by pllim** [comment]
devdeps is unrelated (we have to wait for scikit-image to catch up with numpy)

File: CHANGES.rst [hunk 1]
```diff
@@ -79,6 +79,9 @@ Specviz2d
 Bug Fixes
 ---------
 
+- Circle tool to create circular Subset no longer result in an ellipse
```
Comment (kecnry) [inline CHANGES.rst:82]:
```suggestion
- Circle tool to create a circular Subset no longer results in an ellipse
```

**Review by kecnry — APPROVED** [review]
Thanks!  (I see no reason to backport this - especially since it bumps the pin to bleeding-edge glue-jupyter).

**Review by javerbukh — APPROVED** [review]
LGTM

**Comment by pllim** [comment]
Thanks for the reviews!

**Pull request merged**